script/DeployMaxVault.s.sol

```diff
@@ -148,7 +147,17 @@ contract DeployMaxVault is BaseScript {
             setWithdrawRule(vault, contracts.YNCLISBNBK());
             setDepositAssetRule(vault, contracts.YNCLISBNBK(), contracts.WBNB());
             setWithdrawAssetRule(vault, contracts.YNCLISBNBK(), contracts.WBNB());
-            setApprovalRule(vault, contracts.YNCLISBNBK(), contracts.WBNB());
+        }
+
+        if (contracts.YNWBNBK() != address(0x0b) && contracts.YNCLISBNBK() != address(0x0c)) {
+            address[] memory underlyingVaults = new address[](2);
+            underlyingVaults[0] = contracts.YNWBNBK();
+            underlyingVaults[1] = contracts.YNCLISBNBK();
+            setApprovalRule(vault, contracts.WBNB(), underlyingVaults);
+        } else if (contracts.YNWBNBK() != address(0x0b)) {
+            setApprovalRule(vault, contracts.WBNB(), contracts.YNWBNBK());
+        } else if (contracts.YNCLISBNBK() != address(0x0c)) {
+            setApprovalRule(vault, contracts.WBNB(), contracts.YNCLISBNBK());
```

Comment:
We need to set a single approval rule, since the rules are stored per contractAddress per funcSig, so only a single rule can be set for `contracts.WBNB()` and `approve` funcSig